### DISCUSSION TITLE
Need guidance for generate outer key and use it.

### DISCUSSION THREAD

**Discussion by mvanashri** [dashingsoft/pyarmor#1540] [Q&A]
I am facing one issue like i have encrypted my code and share that to target system. That is working fine. I have generated outer key for protection of code from source machine and provide it to target system. but that i haven't provided to target system. then code was not working. Then I have generated the outer key in target machine itself and used it .The code was working fine.
So my doubt is ,is this ok.the target machine should not able to generate the outer key.

Can anyone please help mi in this.

**Comment by jondy** [comment]
No. You should not install Pyarmor in target machine.

**Reply by mvanashri** [reply]
However, the client has the option to install and use that one right if he discovers that we are obfuscating using Pyarmor.

**Comment by jondy** [comment]
If you're using purchased version, the client can't generate outer key for your obfuscated scripts.
